airflow/models/taskinstance.py

```diff
@@ -552,9 +554,11 @@ def clear_task_instances(
                 dr.state = dag_run_state
                 dr.start_date = timezone.utcnow()
                 if dag_run_state == DagRunState.QUEUED:
-                    dr.last_scheduling_decision = None
+                    dr.activate_scheduling()
                     dr.start_date = None
                     dr.clear_number += 1
+            else:
+                dr.activate_scheduling()
```

Comment:
Wouldn't this mean a running dagrun wouldn't have scheduling turned back on? Should the activate in else just be done for all dagruns in `drs`?